com.unity.render-pipelines.universal/Shaders/Terrain/TerrainDetailLit.shader

```diff
@@ -247,7 +249,7 @@ Shader "Hidden/TerrainEngine/Details/UniversalPipeline/Vertexlit"
                 surfaceData.occlusion = 1.0;
 
                 InputData inputData = (InputData)0;
-                inputData.normalWS = half3(0, 1, 0); // need some default to avoid division by 0.
+                inputData.normalWS = input.NormalWS;
```

Comment:
can input.NormalWS be (0,0,0)? Is the previous comment relevant here?

Reply:
I'll find someone from the Terrain team to look at this for me and confirm that for me.

Reply:
It might make more sense if the consuming code (e.g. in the normalization code) handles the zero length normal since we don't really know here what is coming in or how it gets used in URP.

Alternatively here we can do a length-squared check to see if input.NormalWS is zero length, but that cost only makes sense if we think the consuming code won't already handle that condition.